napari/layers/graph/graph.py

```diff
@@ -0,0 +1,322 @@
+            properties=properties,
+            text=text,
+            symbol=symbol,
+            size=size,
+            border_width=border_width,
+            border_width_is_relative=border_width_is_relative,
+            border_color=border_color,
+            border_color_cycle=border_color_cycle,
+            border_colormap=border_colormap,
+            border_contrast_limits=border_contrast_limits,
+            face_color=face_color,
+            face_color_cycle=face_color_cycle,
+            face_colormap=face_colormap,
+            face_contrast_limits=face_contrast_limits,
+            out_of_slice_display=out_of_slice_display,
+            n_dimensional=n_dimensional,
+            name=name,
+            metadata=metadata,
+            scale=scale,
+            translate=translate,
+            rotate=rotate,
+            shear=shear,
+            affine=affine,
+            opacity=opacity,
+            blending=blending,
+            visible=visible,
+            cache=cache,
+            property_choices=property_choices,
+            experimental_clipping_planes=experimental_clipping_planes,
+            shading=shading,
+            canvas_size_limits=canvas_size_limits,
+            antialiasing=antialiasing,
+            shown=shown,
+        )
+
+    @staticmethod
+    def _fix_data(
+        data: Optional[Union[BaseGraph, ArrayLike]] = None,
+        ndim: Optional[int] = None,
+    ) -> BaseGraph:
+        """Checks input data and return a empty graph if is None."""
+        if ndim is None:
+            ndim = 3
+
+        if data is None:
+            return UndirectedGraph(n_nodes=100, ndim=ndim, n_edges=200)
+
+        if isinstance(data, BaseGraph):
+            if data._coords is None:
+                raise ValueError(
+                    trans._(
+                        "Graph layer must be a spatial graph, have the `coords` attribute."
+                    )
+                )
+            return data
+
+        try:
+            arr_data = np.atleast_2d(data)
+        except ValueError as err:
+            raise NotImplementedError(
+                trans._(
+                    "Could not convert to {data} to a napari graph.",
+                    data=data,
+                )
+            ) from err
+
+        if not issubclass(arr_data.dtype.type, np.number):
+            raise TypeError(
+                trans._(
+                    "Expected numeric type. Found{dtype}.",
+                    dtype=arr_data.dtype,
+                )
+            )
+
+        if arr_data.ndim > 2:
+            raise ValueError(
+                trans._(
+                    "Graph layer only supports 2-dim arrays. Found {ndim}.",
+                    ndim=arr_data.ndim,
+                )
+            )
+        return UndirectedGraph(coords=arr_data)
+
+    @property
+    def _points_data(self) -> np.ndarray:
+        return self._data._coords
+
+    @property
+    def data(self) -> BaseGraph:
```

Comment:
I think this is the first layer type where the `data` getter returns a non-standard type (i.e. not a sequence of arrays). Personally, I think that's OK, but I think it's worth calling out as one of the bigger aspects of the Graph layer type.